openshift_cli_installer/utils/gcp_utils.py

```diff
@@ -0,0 +1,103 @@
+
+
+def get_gcp_params():
+    gcp_params = {}
+    gcp_params["gcp_sa_file_dir"] = os.path.join(os.path.expanduser("~"), ".gcp")
+    gcp_params["openshift_installer_gcp_sa_file_path"] = os.path.join(
+        gcp_params["gcp_sa_file_dir"], "osServiceAccount.json"
+    )
+    gcp_params["backup_existing_gcp_sa_file_path"] = os.path.join("/tmp", f"installerTemp{short_uuid}.json")
+    return gcp_params
+
+
+def set_gcp_configuration(**kwargs):
+    """
+    Saves provided GCP Service Account file at location '~/.gcp/osServiceAccount.json'
+
+    If file already exists at this path, it will be copied to tmp file first.
+
+    """
+    user_input = get_user_input(**kwargs)
+    if is_platform_gcp(user_input["clusters"]):
+        gcp_params = get_gcp_params()
+        if os.path.exists(gcp_params["openshift_installer_gcp_sa_file_path"]):
+            LOGGER.info(
+                f"File {gcp_params['openshift_installer_gcp_sa_file_path']} already exists. Copying to {gcp_params['backup_existing_gcp_sa_file_path']}"
+            )
+            shutil.copy(
+                gcp_params["openshift_installer_gcp_sa_file_path"], gcp_params["backup_existing_gcp_sa_file_path"]
+            )
+        else:
+            Path(gcp_params["gcp_sa_file_dir"]).mkdir(parents=True, exist_ok=True)
+        LOGGER.info(f"Saving GCP ServiceAccount file to {gcp_params['openshift_installer_gcp_sa_file_path']}")
+        shutil.copy(user_input["gcp_service_account_file"], gcp_params["openshift_installer_gcp_sa_file_path"])
+
+
+def restore_gcp_configuration(**kwargs):
+    """
+    Restores location '~/.gcp/osServiceAccount.json'
+
+    Copy file from tmp location to '~/.gcp/osServiceAccount.json' if exists before,
+    Otherwise removes the directory '~/.gcp'
+
+    """
+    user_input = get_user_input(**kwargs)
+    if is_platform_gcp(user_input["clusters"]):
+        gcp_params = get_gcp_params()
+        if os.path.exists(gcp_params["backup_existing_gcp_sa_file_path"]):
+            LOGGER.info(f"Restoring previous file contents of {gcp_params['openshift_installer_gcp_sa_file_path']}")
+            shutil.copy(
+                gcp_params["backup_existing_gcp_sa_file_path"], gcp_params["openshift_installer_gcp_sa_file_path"]
+            )
+            os.remove(gcp_params["backup_existing_gcp_sa_file_path"])
+        else:
+            LOGGER.info(f"Deleting path {gcp_params['openshift_installer_gcp_sa_file_path']}")
+            shutil.rmtree(gcp_params["gcp_sa_file_dir"])
+
+
+def is_platform_gcp(clusters):
+    if any([_cluster["platform"] == GCP_STR for _cluster in clusters]):
+        return True
+
+
+def get_service_account_dict_from_file(gcp_service_account_file):
+    with open(gcp_service_account_file) as fd:
+        return json.loads(fd.read())
+
+
+def get_gcp_project_id(gcp_service_account_file):
+    return get_service_account_dict_from_file(gcp_service_account_file=gcp_service_account_file)["project_id"]
+
+
+def get_user_input(**kwargs):
```

Comment:
@myakove @rnetser updated the code, please review, this function is bit of duplicate of what we do in UserInput class, should we also update [UserInput](https://github.com/RedHatQE/openshift-cli-installer/blob/main/openshift_cli_installer/libs/user_input.py) to call this function instead?

Reply:
In this case, save the user_input return value in `cli.py` line 215 and use it
No need for this function.

Reply:
okay, I'll revert these changes